### ISSUE TITLE
First selected item not announced

### ISSUE THREAD

**Issue by ebonow** [mynamesleon/aria-autocomplete#22]
When using iOS VoiceOver, it seems that the first selected item is not announced.

Here is the captured result of typing a letter and pressing the down arrow
<img width="1274" alt="Screen Shot 2021-01-21 at 10 57 39 AM" src="https://user-images.githubusercontent.com/1007579/105398881-b9e7bb00-5bd7-11eb-98c4-b3227fa1c760.png">

**Comment by mynamesleon** [comment]
Sorry for my delayed reply to this.

I have to admit, the majority of my screen reader testing of this has been done using NVDA and JAWS on Windows, along with additional checks using VoiceOver on Mac, and Android TalkBack. So I haven't tested it extensively on iOS VoiceOver.

I assume you were you using iOS VoiceOver on an iPad judging by the dimensions of your screenshot?

Interestingly, for the item announcements, I'm relying solely on focus states and the WAI-ARIA spec (rather than doing any custom announcements of my own). In this case, each item in the list has `role="option"` set, and when it receives focus the `aria-selected` attribute is updated to `"true"`. VoiceOver has definitely had issues in the past with the `aria-selected` attribute, so I'm inclined to believe this is a VoiceOver issue, but definitely needs investigating.

Could you please clarify a few things for me:

1. Which version of iOS were you using on your device?
2. Was it simply that it wasn't announced, or was it still announcing the help text? 
3. Were later items in the list announced correctly for you then?
4. Was this also from the first search attempt, or a subsequent one?

**Comment by ebonow** [comment]
@mynamesleon 
1. I am using OSX version 10.15.7 (MacBookPro)
2. It wasn't announced on first focus. I attempted to reproduce the issue, and had mixed results so I will close this and perhaps write this off as a VoiceOver anomaly.
3. The other items were announced perfectly as expected
4. First search attempt, subsequent ones 

One other strange behavior I just ran into is that there is a stutter when focusing on options. It appears that the option is first read with the index (ex: Afghanistan (1 of 163)), but then is quickly replaced by just the label of the option (ex: Afghanistan) so the first letter kind of stutters or repeats mid-syllable.

https://user-images.githubusercontent.com/1007579/110152330-352bb780-7da7-11eb-96d3-858402ffebae.mov

**Comment by mynamesleon** [comment]
Regarding the first selected item not being announced: I've re-tested this on MacOS 11 and was unable to reproduce it. I tried using Safari, as well as the latest versions of Chrome, Brave (as an alternative Chromium browser test), and Firefox.

VoiceOver is very aggressive with announcing the full textbox contents once you've finished typing. This often cancels the "n results available" announcement. E.g. after typing "afg", and pausing, the "1 result available" announcement starts, and is then cancelled by a repeat "afg" announcement (this is another issue I'm aware of, and I have experimented with announcement delays in the past to counteract it, with mixed results). I'm wondering if this behaviour might be affecting the announcement of the first item as well. Interestingly in your video clip above, it appears to announce the first item correctly. So this does seem to be a VoiceOver anomaly, particularly as I've not been able to reproduce it on MacOS 11. I might downgrade my MacOS version later to test this further.

Regarding the stutter when focusing options: yes I've experienced this as well. It seems to be an oddity that only applies to Chrome though in my testing - it wasn't happening in Safari or Firefox, or even in the other Chromium based browser (Brave) that I regularly test with. I've noticed it happening in similar combobox examples on other sites as well. So it's definitely a browser specific bug in how Chrome is interacting with VoiceOver's accessibility API (interestingly, if you hold the arrow key to skip over a few options, the announcement is then done currently).
